viewer/packages/utilities/src/events/InputHandler.ts

```diff
@@ -143,3 +146,12 @@ export class InputHandler {
     this._events.hover.fire(clickOrTouchEventOffset(e, this.domElement));
   }, 100);
 }
+
+/**
+ * Method for deleting all external events that are associated with current instance of a class.
+ */
+export function disposeOfAllEventListeners(eventList: any): void {
```

Comment:
Add a proper type for `eventList`. I believe it would be something along the lines of 
```suggestion
export type EventCollection = { [eventName: string]: EventHandler<void> };
export function disposeOfAllEventListeners(eventList: EventCollection): void {
```

Reply:
Seems like it won't work that simple, because eventList has different generic type in each event type field.
It could be `_events = {
    cameraChange: new EventTrigger<CameraChangeDelegate>(),
    click: new EventTrigger<PointerEventDelegate>(),
    hover: new EventTrigger<PointerEventDelegate>(),
    sceneRendered: new EventTrigger<SceneRenderedDelegate>(),
    disposed: new EventTrigger<DisposedDelegate>()
  }`. I don't think you can create a general type for such a thing, because list can also include or exclude some of these types of events.

Reply:
What if you use `EventHandler<unknown>` instaed of `EventHandler<void>`?

Reply:
It says: `Type 'unknown' does not satisfy the constraint '(...args: any[]) => void'.` But with `any` it works fine:
`type EventCollection = { [eventName: string]: EventTrigger<any> };`.

Reply:
I think we should try any then. Or maybe unknown => void works?

Reply:
No, not working like this but another option is just the general type from EventTrigger, i.e. `type EventCollection = { [eventName: string]: EventTrigger<(...args:any) => void> };`. This one also works fine.

Reply:
The more specific the better, so I think the last suggestion you have looks good!

Reply:
(might ber that `...args: unknown[]` works too - thats better than `...args:any[]` if it works)

Reply:
Works in type declaration but it is not automatically assigned to `_events` object, then every time it is required to write `_events as EventCollection`. So, I think any version is better in that case.

Reply:
Added TODO